components/validator/guardian.go

```diff
@@ -0,0 +1,159 @@
+package validator
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	_ "net/http/pprof"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/kroma-network/kroma/bindings/bindings"
+	"github.com/kroma-network/kroma/components/node/eth"
+	"github.com/kroma-network/kroma/utils"
+)
+
+// Guardian is responsible for validating outputs
+type Guardian struct {
+	log      log.Logger
+	cfg      Config
+	ctx      context.Context
+	cancel   context.CancelFunc
+	callOpts *bind.CallOpts
+	txOpts   *bind.TransactOpts
+	wg       sync.WaitGroup
+
+	securityCouncilContract *bindings.SecurityCouncil
+	securityCouncilSub      ethereum.Subscription
+}
+
+// NewGuardian creates a new Guardian
+func NewGuardian(parentCtx context.Context, cfg Config, l log.Logger) (*Guardian, error) {
+	securityCouncilContract, err := bindings.NewSecurityCouncil(cfg.SecurityCouncilAddr, cfg.L1Client)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithCancel(parentCtx)
+
+	return &Guardian{
+		log:                     l,
+		cfg:                     cfg,
+		ctx:                     ctx,
+		cancel:                  cancel,
+		callOpts:                utils.NewCallOptsWithSender(ctx, cfg.TxManager.From()),
+		txOpts:                  utils.NewSimpleTxOpts(ctx, cfg.TxManager.From(), cfg.TxManager.Signer),
+		securityCouncilContract: securityCouncilContract,
+	}, nil
+}
+
+func (g *Guardian) Stop() {
+	g.log.Info("stop Guardian")
+
+	if g.securityCouncilSub != nil {
+		g.securityCouncilSub.Unsubscribe()
+	}
+}
+
+func (g *Guardian) handleValidationRequested(logs chan *bindings.SecurityCouncilValidationRequested) {
+	defer func() {
+		g.log.Info("exit handleRequestValidated goroutine")
+		close(logs)
+		g.wg.Done()
+	}()
+	for {
+		select {
+		case ev := <-logs:
+			if !g.isSecurityCouncil(g.callOpts.From) {
+				continue
+			}
+			g.wg.Add(1)
+			go g.processOutputValidation(ev)
+		case <-g.ctx.Done():
+			return
+		}
+	}
+}
+
+func (g *Guardian) processOutputValidation(event *bindings.SecurityCouncilValidationRequested) {
+	g.log.Info("start processOutputValidation")
+	defer func() {
+		g.log.Info("exit processOutputValidation goroutine")
+		g.wg.Done()
+	}()
+
+	isValidate, err := g.isL2OutputValidate(event.OutputRoot, event.L2BlockNumber.Uint64())
+	if err != nil {
+		g.log.Error("isL2OutputValidate failed", "err", err)
+		return
+	}
```

Comment:
We don't have to retry this output validation when some errors occur (e.g. network err)?